src/core/scope.c

```diff
@@ -416,6 +474,32 @@ static int scope_start(Unit *u) {
         return 1;
 }
 
+static int scope_start(Unit *u) {
+        Scope *s = SCOPE(u);
+
+        assert(s);
+
+        if (unit_has_name(u, SPECIAL_INIT_SCOPE))
+                return -EPERM;
+
+        if (s->state == SCOPE_FAILED)
+                return -EPERM;
+
+        /* We can't fulfill this right now, please try again later */
+        if (IN_SET(s->state, SCOPE_STOP_SIGTERM, SCOPE_STOP_SIGKILL))
+                return -EAGAIN;
+
+        assert(s->state == SCOPE_DEAD);
+
+        if (!u->transient && !MANAGER_IS_RELOADING(u->manager))
+                return -ENOENT;
+
+        if (s->user)
```

Comment:
This is simple(r). Is the omission of `s->group` intentional?

Reply:
Actually no, this is on purpose. This works the same way for services, i.e. having 'Delegate=yes Group=foo' w/o specifing User= has no effect.

```shell
$ useradd test

# w/o User= the delegation is not enabled
$ systemd-run --unit test.service --wait -t -p Delegate=yes -p Group=test ls -ld /sys/fs/cgroup/system.slice/test.service 2>/dev/null
drwxr-xr-x 2 root root 0 Jun  9 13:12 /sys/fs/cgroup/system.slice/test.service

# With User= the delegation is set up
$ systemd-run --unit test.service --wait -t -p Delegate=yes -p User=test -p Group=test ls -ld /sys/fs/cgroup/system.slice/test.service 2>/dev/null
drwxr-xr-x 2 test test 0 Jun  9 13:12 /sys/fs/cgroup/system.slice/test.service
```

Maybe this behavior is unexpected, but for start I wanted scopes to behave the same way as services do.

Reply:
Thanks for your example. Please add a comment about consistency with services (however unexpected it is).

(FTR, "Delegate=yes Group=foo' w/o specifing User=" still has effect on what cgroup controllers are enabled in the subtree. There's no effect on ownership indeed.)